src/core/Akka/Util/ByteString.cs

```diff
@@ -28,6 +29,46 @@ namespace Akka.IO
     [DebuggerDisplay("(Count = {_count}, Buffers = {_buffers})")]
     public sealed class ByteString : IEquatable<ByteString>, IEnumerable<byte>
     {
+        public class ByteStringReadOnlySequenceSegment : ReadOnlySequenceSegment<byte>
```

Comment:
Not sure where to put in class (or maybe even move to partial/etc?)

It's a nested class for the sake of being able to directly access `_buffers` in the create. OTOH if we 'flipped' the construction (see other comment) we wouldn't need to nest it and could just build in `.ToReadOnlySequence()`.

... Either way this should be `sealed` though